app/src/main/java/org/dicekeys/app/views/DiceKeyView.kt

```diff
@@ -101,22 +119,33 @@ class DiceKeyView @JvmOverloads constructor(
 
             canvas.save()
             canvas.translate(sizeModel.marginLeft, sizeModel.marginTop)
-            for (facePosition in facePositions) {
+            for ((i, facePosition) in facePositions.withIndex()) {
                 canvas.save()
                 canvas.translate(
                         dieStepSize * facePosition.column,
                         dieStepSize * facePosition.row
                 )
-                if (computedShowDiceAtIndexes.contains(facePosition.id)) {
+
+                val inCenter = (i == 12)
+
+                 if(computedShowDiceAtIndexes.contains(facePosition.id) && ((facesAreHidden && inCenter) || !facesAreHidden)){
```

Comment:
The center face feature is being re-implemented here when it already exists in computedShowDiceAtIndexes. If we get rid of that (I think the product is far enough along that we can be confident we'll either show all dice or the center)... I think we can simplify this logic to:

```
if((inCenter || !facesAreHidden)){
```

If you rename `inCenter` to `dieIsCenterDie` and `facesAreHidden` to `hideDiceExceptCenterDie`, and re-order the clauses, then it becomes even more readable
```
val dieIsCenterDie = (i == 12)
if (hideDiceExceptCenterDie && !dieIsCenterDie)){
  canvas.drawRoundRect(0f, 0f, faceSize, faceSize,sizeModel.faceRadius, sizeModel.faceRadius, diceBoxDieSlotPaint)
}else{
  val dieFace = facePosition.drawable
  dieFace.highlighted = highlightedIndexes.contains(facePosition.id)
  dieFace.draw(canvas)
}
```

Reply:
Applied your suggestion and clean up the code.